From cf6f2c635394aab9c64f16116fe970ec403cd6cf Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 25 Jul 2018 13:44:20 +0800
Subject: [PATCH 4395/5242] MLK-19042-6: ARM64: dts: enable sound-rpmsg in
 fsl-imx8mm-evk-m4

commit  f30a02522e446c059d4c719f9420371a3761dbe2 from
https://source.codeaurora.org/external/imx/linux-imx.git

enable sound-rpmsg in fsl-imx8mm-evk-m4

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
(cherry picked from commit 195a79896f105e5537ae4dfde97e87a9e198cbbd)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-imx8mm-evk-m4.dts |   31 ++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-m4.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-m4.dts
index 2501937..3e68451 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-m4.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-m4.dts
@@ -30,6 +30,25 @@
 	sound-wm8524 {
 		status = "disabled";
 	};
+
+	wm8524: wm8524 {
+		status = "disabled";
+	};
+
+	rpmsg_i2s: rpmsg-i2s {
+		compatible = "fsl,imx8mq-rpmsg-i2s";
+		/* the audio device index in m4 domain */
+		fsl,audioindex = <0> ;
+		fsl,dma-buffer-size = <0x6000000>;
+		status = "okay";
+	};
+
+	sound-rpmsg {
+		compatible = "fsl,imx-audio-rpmsg";
+		model = "rpmsg-audio";
+		cpu-dai = <&rpmsg_i2s>;
+	};
+
 };
 
 /*
@@ -48,3 +67,15 @@
 &uart4 {
 	status = "disabled";
 };
+
+&sai3 {
+	status = "disabled";
+};
+
+&sdma3 {
+	status = "disabled";
+};
+
+&sai1 {
+	status = "disabled";
+};
-- 
1.7.9.5

